core/src/test/java/io/grpc/ChannelImplTest.java

```diff
@@ -0,0 +1,262 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.same;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import io.grpc.transport.ClientStream;
+import io.grpc.transport.ClientStreamListener;
+import io.grpc.transport.ClientTransport;
+import io.grpc.transport.ClientTransportFactory;
+
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/** Unit tests for {@link ChannelImpl}. */
+@RunWith(JUnit4.class)
+public class ChannelImplTest {
+  private MethodDescriptor<String, Integer> method = MethodDescriptor.create(
+      MethodType.UNKNOWN, "/service/method", 1, TimeUnit.SECONDS,
+      new StringMarshaller(), new IntegerMarshaller());
+  private ExecutorService executor = Executors.newSingleThreadExecutor();
+  private ClientTransportFactory mockTransportFactory = mock(ClientTransportFactory.class);
+  private ChannelImpl channel = new ChannelImpl(mockTransportFactory, executor);
+  @SuppressWarnings("unchecked")
+  private ClientCall.Listener<Integer> mockCallListener = mock(ClientCall.Listener.class);
+  @SuppressWarnings("unchecked")
+  private ClientCall.Listener<Integer> mockCallListener2 = mock(ClientCall.Listener.class);
+  @SuppressWarnings("unchecked")
+  private ClientCall.Listener<Integer> mockCallListener3 = mock(ClientCall.Listener.class);
+  private ArgumentCaptor<ClientTransport.Listener> transportListenerCaptor =
```

Comment:
@Captor isn't actually ever needed for working around Generics issues. [You can pass `null` to ArgumentCaptor](https://github.com/grpc/grpc-java/blob/a251171850fcc1fe6873b0560e38d399f0ac69ab/stub/src/test/java/io/grpc/stub/ClientCallsTest.java#L71) without any issues.

ArgumentCaptor should really just have a zero-arg version in Mockito. ArgumentCaptor explicitly states that it doesn't use the provided class.